internal/dag/httpproxy_processor.go

```diff
@@ -297,6 +297,14 @@ func (p *HTTPProxyProcessor) computeHTTPProxy(proxy *contour_api_v1.HTTPProxy) {
 				} else {
 					svhost.AuthorizationResponseTimeout = timeout
 				}
+
+				if auth.BufferSettings != nil {
+					svhost.AuthorizationServerBufferSettings = &AuthorizationServerBufferSettings{
+						MaxRequestBytes:     auth.BufferSettings.MaxRequestBytes,
```

Comment:
unfortunately we can't just pass this straight through, in the case that someone makes an httpproxy that looks like:

```
  virtualhost:
    authorization:
      authPolicy:
        context:
          hostname: externalauth.projectcontour.io
      bufferSettings: {}
```

we get `0` for this value since it is unset, and Envoy returns the following xDS error:

```
time="2021-11-15T16:46:46Z" level=error msg="Error adding/updating listener(s) ingress_https: Proto constraint validation failed (ExtAuthzValidationError.WithRequestBody: embedded message failed validation | caused by BufferSettingsValidationError.MaxRequestBytes: value must be greater than 0): grpc_service {\n  envoy_grpc {\n    cluster_name: \"extension/httpproxy-external-auth/testserver\"\n  }\n  timeout {\n    nanos: 500000000\n  }\n}\nwith_request_body {\n}\nclear_route_cache: true\nstatus_on_error {\n  code: Forbidden\n}\ninclude_peer_certificate: true\ntransport_api_version: V3\n\n" code=13 connection=2 context=xds node_id=envoy-k7l9l node_version=v1.19.1 response_nonce=4 version_info=3
```

we may have to set a default in kubebuilder and/or validate the field is not in fact `0` in here and set a default size

Reply:
When the field is unsetin YAML it will still come through as 0, in the example above you can see that if someone uses {} for the buffersettings

Reply:
Yes, I think the way to think this through is: what happens when someone supplies the absolute minimum of configuration? Does the combination of just-the-required-fields and zero values for optional fields work?

Reply:
@sunjayBhatia Thanks for the clarification. Added` validation check. 

The other 2 fields are boolean; so we just need to take care of validating and setting default for maxBytes.